tamr_toolbox/realtime/matching.py

```diff
@@ -0,0 +1,237 @@
+import json
+import logging
+import requests
+import time
+from collections import defaultdict
+from typing import Dict, List, Optional, Union
+
+from tamr_unify_client import Client
+from tamr_unify_client.mastering.project import MasteringProject
+from tamr_unify_client.operation import Operation
+from tamr_toolbox.utils.operation import from_resource_id
+from tamr_toolbox.models.data_type import JsonDict
+
+LOGGER = logging.getLogger(__name__)
+
+
+def update_realtime_match_data(
+    *,
+    project: MasteringProject,
+    do_update_clusters: bool = True,
+    do_use_manual_clustering: bool = False,
+    **options,
+) -> Operation:
+    """
+    Updates data for RealTime match queries if needed, based on latest published clusters.
+
+    Args:
+        project: project to be updated
+        do_update_clusters: whether to update clusters, default True
+        do_use_manual_clustering: whether to use externally managed clustering, default False
+        options: Options passed to underlying :class:`~tamr_unify_client.operation.Operation`
+    """
+
+    # Make sure we have the original name of the project to use with the match endpoint
+    project_name = _get_internal_project_name(project.client, project_id=project.resource_id)
+
+    url = (
+        f"projects/{project_name}:updateLLM?updateClusters={do_update_clusters}"
+        f"&useManualClustering={do_use_manual_clustering}"
+    )
+    response = project.client.post(url)
+    if not response.ok:
+        message = (
+            f"Match data update for {project_name} failed at submission time: "
+            + response.json()["message"]
+        )
+        LOGGER.error(message)
+        raise RuntimeError(message)
+    operation_id = response.content.decode("latin1")
+    operation = from_resource_id(project.client, job_id=operation_id)
+
+    return operation.apply_options(**options)
+
+
+def poll_realtime_match_status(
```

Comment:
This function and `match_query` should take in a `Project` object instead of project name.  Then within, your `_get_internal_project_name` function should be used to get the name needed for constructing the request URL.

Reply:
I have the same concern, but this pattern is going to be necessary since these endpoints need to have the internal project name (name at creation time).  We can migrate the internals for Tamr v2022.004+ to use the project ID, but the `Project` object is the best way to maintain compatibility across versions.